agent-providers/span/src/main/java/com/expedia/www/haystack/agent/pitchfork/service/PitchforkServlet.java

```diff
@@ -69,17 +70,19 @@ protected void doPost(final HttpServletRequest request, final HttpServletRespons
                 response.sendError(503, "Fail to process/forward the zipkin span!");
             }
         } else {
-            response.sendError(400, String.format("invalid content-type, supported values are %s, %s, %s",
-                    JSON_CONTENT_TYPE, THRIFT_CONTENT_TYPE, PROTO_CONTENT_TYPE));
+            response.sendError(400, String.format("invalid content-type, supported values are %s, %s, %s, got '%s'",
+                    JSON_CONTENT_TYPE, THRIFT_CONTENT_TYPE, PROTO_CONTENT_TYPE, request.getContentType()));
         }
     }
 
     private ZipkinSpanProcessor getProcessor(String contentType) {
-        final String[] contentTypes = contentType.split(";");
-        for (final String ctype : contentTypes) {
-            final ZipkinSpanProcessor processor = processors.get(ctype.toLowerCase());
-            if (processor != null) {
-                return processor;
+        if(!isEmpty(contentType) ) {
```

Comment:
What about using early termination to avoid identation levels?

Reply:
I don't know if there is a style preference from the owners of this project.
If no one else comments I'll do that change :)